zwave_js_server/exceptions.py

```diff
@@ -63,6 +64,19 @@ class InvalidMessage(BaseZwaveJSServerError):
 class InvalidServerVersion(BaseZwaveJSServerError):
     """Exception raised when connected to server with incompatible version."""
 
+    def __init__(
+        self,
+        version_info: Optional["VersionInfo"],
+        required_schema_version: int,
+        message: str,
+    ) -> None:
+        """Initialize an invalid server version error."""
+        assert version_info
```

Comment:
We should not accept an optional version if that's not ok. Make the callers check that they pass the correct argument instead.